packages/dd-trace/test/llmobs/tagger.spec.js

```diff
@@ -0,0 +1,430 @@
+      }
+    }
+
+    util = {
+      generateTraceId: sinon.stub().returns('0123')
+    }
+
+    logger = {
+      warn: sinon.stub()
+    }
+
+    Tagger = proxyquire('../../src/llmobs/tagger', {
+      '../log': logger,
+      './util': util
+    })
+
+    tagger = new Tagger({ llmobs: { mlApp: 'my-default-ml-app' } })
+  })
+
+  describe('setLLMObsSpanTags', () => {
+    it('tags an llm obs span with basic and default properties', () => {
+      tagger.setLLMObsSpanTags(span, 'workflow')
+
+      expect(span.context()._tags).to.deep.equal({
+        'span.type': 'llm',
+        '_ml_obs.meta.span.kind': 'workflow',
+        '_ml_obs.meta.ml_app': 'my-default-ml-app',
+        '_ml_obs.llmobs_parent_id': 'undefined' // no parent id provided
+      })
+    })
+
+    it('uses options passed in to set tags', () => {
+      tagger.setLLMObsSpanTags(span, 'llm', {
+        modelName: 'my-model',
+        modelProvider: 'my-provider',
+        sessionId: 'my-session',
+        mlApp: 'my-app'
+      })
+
+      expect(span.context()._tags).to.deep.equal({
+        'span.type': 'llm',
+        '_ml_obs.meta.span.kind': 'llm',
+        '_ml_obs.meta.model_name': 'my-model',
+        '_ml_obs.meta.model_provider': 'my-provider',
+        '_ml_obs.session_id': 'my-session',
+        '_ml_obs.meta.ml_app': 'my-app',
+        '_ml_obs.llmobs_parent_id': 'undefined'
+      })
+    })
+
+    it('uses the name if provided', () => {
+      tagger.setLLMObsSpanTags(span, 'llm', {}, 'my-span-name')
+
+      expect(span.context()._tags).to.deep.equal({
+        'span.type': 'llm',
+        '_ml_obs.meta.span.kind': 'llm',
+        '_ml_obs.meta.ml_app': 'my-default-ml-app',
+        '_ml_obs.llmobs_parent_id': 'undefined',
+        '_ml_obs.name': 'my-span-name'
+      })
+    })
+
+    it('defaults parent id to undefined', () => {
+      tagger.setLLMObsSpanTags(span, 'llm')
+
+      expect(span.context()._tags).to.deep.equal({
+        'span.type': 'llm',
+        '_ml_obs.meta.span.kind': 'llm',
+        '_ml_obs.meta.ml_app': 'my-default-ml-app',
+        '_ml_obs.llmobs_parent_id': 'undefined'
+      })
+    })
+
+    it('uses the parent span if provided to populate fields', () => {
+      const parentSpan = {
+        context () {
+          return {
+            _tags: {
+              '_ml_obs.meta.ml_app': 'my-ml-app',
+              '_ml_obs.session_id': 'my-session'
+            },
+            toSpanId () { return '5678' }
+          }
+        }
+      }
+      tagger.setLLMObsSpanTags(span, 'llm', { parentLLMObsSpan: parentSpan })
+
+      expect(span.context()._tags).to.deep.equal({
+        'span.type': 'llm',
+        '_ml_obs.meta.span.kind': 'llm',
+        '_ml_obs.meta.ml_app': 'my-ml-app',
+        '_ml_obs.session_id': 'my-session',
+        '_ml_obs.llmobs_parent_id': '5678'
+      })
+    })
+
+    it('uses the propagated trace id if provided', () => {
+      spanContext._trace.tags['_dd.p.llmobs_trace_id'] = '-123'
```

Comment:
Are we asserting trace ID here? I can't see it 😕

Reply:
oops yeah it's because i removed the test for now but didn't remove this :p good catch!